Jim-Hodapp-Coaching/refactor-platform-fe
# Fetch coaching sessions for selecting the coaching session to join

## Description
Adds a new CoachingSession type to reflect the backend Rust type and fetches 2 months (1 back, 1 forward) worth of sessions and display in the dashboard's select component. Allows the user to select a session to join.

**Note:** This PR requires the [following backend changes](https://github.com/Jim-Hodapp-Coaching/refactor-platform-rs/pull/50) to function correctly.


#### GitHub Issue: N/A

### Changes
* Adds a new CoachingSession type that reflects the backend Rust type
* Fetches 2 months (1 back, 1 forward from now) worth of sessions
* Display these sessions in the select component on the dashboard page
* Allows the user to select and then join a valid coaching session (navigates to that page)

### Screenshots / Videos Showing UI Changes (if applicable)

Dashboard with coaching sessions displayed:
<img width="1680" alt="Screenshot 2024-06-30 at 20 59 26" src="https://github.com/Jim-Hodapp-Coaching/refactor-platform-fe/assets/3219120/cbb8557c-9572-4912-9246-f8b4a9d6bc89">

Coaching sessions page with endpoint shown in browser address bar:
<img width="1680" alt="Screenshot 2024-06-30 at 21 00 47" src="https://github.com/Jim-Hodapp-Coaching/refactor-platform-fe/assets/3219120/582567c3-e581-4ba7-81b9-d3efa9d26933">

### Testing Strategy
1. Log in and the dashboard will show
2. Select an organization
3. Notice that one list of coaching relationships loads
4. Select a coaching relationship
5. Notice that a list of coaching sessions loads, or if there aren't any, "None found" displays
6. With a valid coaching session chosen, notice that the `Join` button enables
7. Click the `Join` button and notice that you're navigated to the `/coaching-sessions/<uuid>` and the coaching session page renders


### Concerns
* Purposefully keeping this PR simpler by not storing all of the dashboard's chosen UUIDs in a Zustand store. This will come in a future PR.
